### PR TITLE
Convert SkipLinks component to Hooks

### PR DESCRIPTION
**Pull request by netochaves**
#### What does this PR do?

Convert SkipLinks Component to Hooks

#### Where should the reviewer start?
`src/js/components/SkipLinks/SkipLinks.js`

#### What testing has been done on this PR?
`yarn test`
#### How should this be manually tested?
`yarn test`
#### Any background context you want to provide?
N/A
#### What are the relevant issues?
#3394
#### Screenshots (if appropriate)
N/A
#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Backward compatible

### CODE REVIEW

File: src/js/components/SkipLinks/SkipLinks.js [hunk 1]
```diff
@@ -1,71 +1,64 @@
-import React, { createRef, cloneElement, Component } from 'react';
+import React, { useRef, useState, cloneElement } from 'react';
```
Comment (ShimiSun) [inline src/js/components/SkipLinks/SkipLinks.js]:
```suggestion
import React, { cloneElement, useRef, useState } from 'react';
```

File: src/js/components/SkipLinks/SkipLinks.js [hunk 2]
```diff
@@ -1,71 +1,64 @@
...
-  };
+const SkipLinks = ({ children, id, messages }) => {
+  const [showLayer, setShowLayer] = useState(false);
+
+  const layerRef = useRef(null);
 
-  state = {
-    showLayer: false,
+  const onFocus = () => {
+    setShowLayer({ showLayer: true });
```
Comment (ShimiSun) [inline src/js/components/SkipLinks/SkipLinks.js]:
```suggestion
    setShowLayer( true );
```
Using hooks allows this further cleanup.

File: src/js/components/SkipLinks/SkipLinks.js [hunk 3]
```diff
@@ -1,71 +1,64 @@
...
   };
 
-  layerRef = createRef();
+  const removeLayer = () => {
+    setShowLayer({ showLayer: false });
```
Comment (ShimiSun) [inline src/js/components/SkipLinks/SkipLinks.js]:
```suggestion
    setShowLayer( false );
```
Using hooks allows this further cleanup.

Comment (netochaves) [reply src/js/components/SkipLinks/SkipLinks.js]: Yeah, thanks.

**Comment by ShimiSun** [comment]
This PR is throwing warnings on the tests,
```
 PASS  src/js/components/SkipLinks/__tests__/SkipLink-test.js
  ● Console

    console.error node_modules/react-dom/cjs/react-dom.development.js:545
      Warning: An update to SkipLinks inside a test was not wrapped in act(...).
      
      When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
          in SkipLinks
          in div (created by Context.Consumer)
          in StyledComponent (created by StyledGrommet)
          in StyledGrommet (created by Grommet)
          in Grommet
```
@netochaves would you like to take a look?

**Comment by netochaves** [comment]
Hey @ShimiSun I've made some research and got figure out that the cause of warning is this  `setTimeout`
https://github.com/grommet/grommet/blob/7efd28655787d4f8aefc2ffaa584227db3c7855a/src/js/components/SkipLinks/SkipLinks.js#L22
Because the state is updated inside the setTimeout wich is async. To remove the warning we have to put the function that call the setTimeout inside an `act` but in react 16.8, `act` don't support async functions only in 16.9 (https://reactjs.org/blog/2019/08/08/react-v16.9.0.html#async-act-for-testing). So i think that only when grommet is updated to react 16.9, we can remove this warning.
More info here: https://github.com/facebook/react/issues/14769

**Comment by ShimiSun** [comment]
Thanks for researching and communicating this detailed answer @netochaves ! I'll take it from here!  Bests

**Pull request merged**